### PR TITLE
feat(wind): Add power law interpolation method for wind conversion

### LINKED ISSUE #231: Use power low instead of log law for extrapolating wind speed
Use power low instead of log law for extrapolating wind speed to hub height.

## Description
The `extrapolate_wind_speed` function uses the power law and the grid averages forecast surface roughness (fsr):
 `wnd_spd = ds[from_name] * (   np.log(to_height / ds["roughness"]) / np.log(from_height / ds["roughness"])  )`
Using the ERA5 fsr with the log law results in too large correction, especially in the Nordics where the fsr is often high.  
At height above the blending height it is better to use the power law. That looks like follow:
`wnd_spd = ds["wnd100m"] * (to_height / 100)**a`. 
The wind sheer exponent `a` can be derived from the wind speed at 10m and 100m: 
`a = np.log(ds["wnd10m"] / ds["wnd100m"]) /  np.log(10 / 100)`. 
Tested it and it give same results as doing actual vertical interpolations with MetPy using the wind components from the pressure level data (how the ERA5 wind at 100 meters is orginally calculated as well).  

For turbines hub heights lower than the blending height it could be useful to use the log law to account for site specific roughness length. Below the equations from the IFS CY41R2 docs used to derive the 10m wind speed in ERA5 (always calculated assuming fsr of 0.03 to better represent conditions at weather stations)

image
The IFS docs mention two blending heights, 40 and 75 meters (I guess it actually might be dynamic and site specific). But a procedure I use my self is to first use the above-mentioned power law to calculate a wind speed at a blending height of 70 meters, then the log law using a site specific fsr (for now, just some blunt estimates based on the ERA5 grid average fsr)

### PR DESCRIPTION
**Pull request by coroa**
Closes #231 .

Refer to issue for context. Implements the power law interpolation method as a new argument
to `cutout.wind`. Needs to be chosen manually for now.

A suggested in the issue we retrieve 100m and 10m wind speeds and then calculate a wind shear exponent from those during cutout preparation.

Looks like the power law is preferred at least for turbine heights above wind blending (?) height (70m?). So this should most likely become the new default if we have 100m wind speeds available (as we have for ERA5). 

TODOs:
- [ ] Compare results, maybe following methodology in https://www.mdpi.com/1996-1073/14/14/4169
- [ ] Implement a FutureWarning that the default changes to the power law.
- [ ] Add those interpolation options to the docs.

## Checklist

- [ ] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [x] Unit tests for new features were added (if applicable).
- [x] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).
- [ ] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [x] I consent to the release of this PR's code under the MIT license.

### CODE REVIEW

**Comment by coroa** [comment]
Tests work locally, any pointers to what is going wrong in CI? @lkstrp

**Comment by lkstrp** [comment]
This PR doesn't have permissions to get the CDS API secret because it comes from a fork, which causes these errors. We might want to change the trigger, or add a manual trigger that has permissions. But the Cutouts should also just be fully cached.

File: atlite/convert.py [hunk 1]
```diff
@@ -529,6 +538,9 @@ def wind(cutout, turbine, smooth=False, add_cutout_windspeed=False, **params):
         output at the highest wind speed in the power curve. If False, a warning will be
         raised if the power curve does not have a cut-out wind speed. The default is
         False.
+    interpolation_method : {"logarithmic", "power"}
+        Law to interpolate wind speed to turbine hub height. Refer to docstring of
+        atlite.wind.extrapolate_wind_speed .
```
Comment (lkstrp) [inline atlite/convert.py:543]:
```suggestion
        Law to interpolate wind speed to turbine hub height. Refer to 
        :py:func:`atlite.wind.extrapolate_wind_speed`.
```

File: atlite/wind.py [hunk 2]
```diff
@@ -13,14 +13,18 @@
...
+):
     """
     Extrapolate the wind speed from a given height above ground to another.
 
     If ds already contains a key refering to wind speeds at the desired to_height,
     no conversion is done and the wind speeds are directly returned.
 
-    Extrapolation of the wind speed follows the logarithmic law as desribed in [1].
+    Extrapolation of the wind speed can either use the "logarithmic" law as
+    described in [1] or the "power" law as described in [2]. See also discussion
```
Comment (lkstrp) [inline atlite/wind.py:26]:
```suggestion
    described in [1]_ or the "power" law as described in [2]_. See also discussion
```

File: atlite/wind.py [hunk 3]
```diff
@@ -47,9 +53,9 @@ def extrapolate_wind_speed(ds, to_height, from_height=None):
     time series from a new global renewable energy atlas for energy system
     analysis'.
```
Comment (lkstrp) [inline atlite/wind.py:54]:
```suggestion
    .. [1] Equation (2) in Andresen, G. et al (2015): 'Validation of Danish wind
       time series from a new global renewable energy atlas for energy system
       analysis'.
```

File: atlite/wind.py [hunk 4]
```diff
@@ -47,9 +53,9 @@ def extrapolate_wind_speed(ds, to_height, from_height=None):
...
 
-    [2] https://en.wikipedia.org/w/index.php?title=Roughness_length&oldid=862127433,
-    Retrieved 2019-02-15.
-
+    [2] Gualtieri, G. (2021): 'Reliability of ERA5 Reanalysis Data for Wind
+    Resource Assessment: A Comparison against Tall Towers'
+    https://doi.org/10.3390/en14144169 .
```
Comment (lkstrp) [inline atlite/wind.py:58]:
```suggestion
    .. [2] Gualtieri, G. (2021): 'Reliability of ERA5 Reanalysis Data for Wind
       Resource Assessment: A Comparison against Tall Towers'
       https://doi.org/10.3390/en14144169 .
```

**Review by lkstrp — APPROVED** [review]
In `pypsa` and `linopy` we now use static types. So always feel free to spam those in there if you touch anything. Long way to go for `atlite` though.

And great to have you back 🎉

File: atlite/datasets/era5.py [hunk 5]
```diff
@@ -44,7 +44,7 @@ def nullcontext():
 
 features = {
     "height": ["height"],
-    "wind": ["wnd100m", "wnd_azimuth", "roughness"],
+    "wind": ["wnd100m", "wnd_shear_exp", "wnd_azimuth", "roughness"],
```
Comment (fneum) [inline atlite/datasets/era5.py:47]: Any problems if the new extrapolation functions are used with an old cutout?

**Review by fneum — COMMENTED** [review]
I like it!

**Comment by coroa** [comment]
> In `pypsa` and `linopy` we now use static types. So always feel free to spam those in there if you touch anything. Long way to go for `atlite` though.

Added quite some typing to the wind conversion now. Needs `typing-extensions` and `from __future__ import annotations` though.

Hope that is agreed practice.

File: atlite/datasets/era5.py (hunk 5 above)
Comment (coroa) [reply atlite/datasets/era5.py:47]: You would raise a `KeyError` when `atlite.wind.extrapolate_wind_speed` tries to access `ds["wnd_shear_exp"]`.

We could raise a more informative error message instead, i guess.

**Comment by lkstrp** [comment]
> Added quite some typing to the wind conversion now. Needs `typing-extensions` and `from __future__ import annotations` though.
> 
> Hope that is agreed practice.

Great, it is. The `__future__` import is only needed for Python `3.9`, and we can remove it once we deprecate `3.9` (and  support `3.13` instead).

File: atlite/datasets/era5.py (hunk 5 above)
Comment (coroa) [reply atlite/datasets/era5.py:47]: Done now

**Comment by coroa** [comment]
> > Added quite some typing to the wind conversion now. Needs `typing-extensions` and `from __future__ import annotations` though.
> > Hope that is agreed practice.
> 
> Great, it is. The `__future__` import is only needed for Python `3.9`, and we can remove it once we deprecate `3.9` (and support `3.13` instead).

Yeah, well, i am confused about the state of the annotations :), since I started reading into https://docs.python.org/3.11/whatsnew/3.11.html#pep-563-may-not-be-the-future .

**Review by fneum — APPROVED** [review]
LGTM.

Side note: Would it make sense to at some point split the `wind_test` into multiple smaller tests?

**Pull request merged**